Src/CSharpier/CommandLineFormatter.cs

```diff
@@ -349,6 +353,35 @@ private bool ShouldIgnoreFile(string filePath)
                 || this.IgnoreFile.IsIgnored(filePath);
         }
 
+        private void WriteError(string filePath, string value)
+        {
+            this.WriteMessage(filePath, value, "Error", ConsoleColor.DarkRed);
+        }
+
+        private void WriteWarning(string filePath, string value)
```

Comment:
You know, have you considered using an `ILogger`? 
https://docs.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line

The more complex this gets, the more it feels like you're re-implementing the wheel. :smirk: